### DISCUSSION TITLE
Best practice for adding a custom entry point e.g. background worker

### DISCUSSION THREAD

**Discussion by jharding** [vercel/next.js#24530] [Help]
I have an application that requires a background worker process to ingest messages from a queue and act accordingly. Since this needs to be a long-running process, it's not compatible with the Next.js framework. That's perfectly fine – I don't expect Next.js to solve every use case for me. However, I want this background worker to share code with the Next.js app and ideally, I'd like it to be integrated into the Next.js build step.

To achieve this currently, I'm injecting an additional entry point into the webpack config in _next.config.js_:

```js
module.exports = {
  webpack: (config, options) => {
    if (options.isServer) {
      const originalEntry = config.entry;

      config.entry = async (...args) => {
        const entry = await originalEntry(...args);

        entry.worker = "./src/worker/index.ts";

        return entry;
      };
    }
  // ...
  }
}
```

I then can execute `node .next/server/worker.js` to start my background worker. This works, but it does feel like I'm reaching into the internals of Next.js a bit and worry that this is potentially a fragile solution.

Has anyone else tried to achieve something similar to this? If so, did you follow a similar pattern? If you did something different, I'd love to hear about it.

**Comment by akomm** [comment]
Does the creation of a custom server file already disable the said features and optimization, or only when you start mess around with routing, etc? Is it possible to leave all the defaults in custom server.js? Then it would be possible to start, say, a worker before starting the server and leave everything at defaults on the next side.

I think the problem with such "side-effects" added are, when you have things run in worker on which static generation depends. The proper approach would be to have a separate "service" running independent. But I get the problem, when you want to use the app's code, so you don't have to either duplicate things or write a separate webpack config to build a service sharing parts of app's code - it smells.
